tests/integration/learning/test_discovery_phases.py

```diff
@@ -101,33 +99,40 @@ def test_alice_verifies_ursula_just_in_time(fleet_of_highperf_mocked_ursulas,
     not_public_key_record_field = RecordField(NotAPublicKey, encode=bytes,
                                               decode=NotAPublicKey.from_bytes)
 
-    _umbral_pubkey_from_bytes = PublicKey.from_bytes
-
-    def actual_random_key_instead(*args, **kwargs):
-        _previous_bytes = args[0]
-        serial = _previous_bytes[-5:]
-        pubkey = NotAPublicKey(serial=serial)
-        return pubkey
-
     def mock_set_policy(id_as_hex):
         return ""
 
     def mock_receive_treasure_map():
         return Response(bytes(), status=201)
 
-    with NotARestApp.replace_route("receive_treasure_map", mock_receive_treasure_map):
-        with NotARestApp.replace_route("set_policy", mock_set_policy):
-            with patch('umbral.PublicKey.__eq__', lambda *args, **kwargs: True):
-                with patch('umbral.PublicKey.from_bytes',
-                           new=actual_random_key_instead):
-                    with patch("nucypher.datastore.models.PolicyArrangement._alice_verifying_key",
-                               new=not_public_key_record_field):
-                        with mock_cert_loading, mock_metadata_validation, mock_message_verification:
-                            with mock_secret_source():
-                                policy = highperf_mocked_alice.grant(
-                                    highperf_mocked_bob, b"any label", m=20, n=30,
-                                    expiration=maya.when('next week'),
-                                    publish_treasure_map=False)
+    def mock_encrypt(public_key, plaintext):
+        if not isinstance(public_key, PublicKey):
+            public_key = public_key.i_want_to_be_a_real_boy()
+        return encrypt(public_key, plaintext)
+
+    mocks = (
+        NotARestApp.replace_route("receive_treasure_map", mock_receive_treasure_map),
+        NotARestApp.replace_route("set_policy", mock_set_policy),
+        patch('nucypher.crypto.umbral_adapter.PublicKey.__eq__', lambda *args, **kwargs: True),
```

Comment:
We no longer need to scope these patches? That is now handled by the code in the `fleet_of_highperf_mocked_ursulas` fixture?

Reply:
They are still applied in the same way as before, I just wanted to flatten 10 nested `with`s - that's what `ExitStack` is for.

Reply:
👍